cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -183,20 +183,17 @@ private Map<String, Set<String>> getThrottledReplicasByTopic(List<ExecutionPropo
     return throttledReplicasByTopic;
   }
 
-  private void setThrottledRateIfUnset(int brokerId) throws ExecutionException, InterruptedException, TimeoutException {
+  private void setThrottledRateIfNecessary(int brokerId) throws ExecutionException, InterruptedException, TimeoutException {
     if (_throttleRate == null) {
       throw new IllegalStateException("Throttle rate cannot be null");
     }
     Config brokerConfigs = getBrokerConfigs(brokerId);
     List<AlterConfigOp> ops = new ArrayList<>();
     for (String replicaThrottleRateConfigKey : Arrays.asList(LEADER_THROTTLED_RATE, FOLLOWER_THROTTLED_RATE)) {
       ConfigEntry currThrottleRate = brokerConfigs.get(replicaThrottleRateConfigKey);
-      if (currThrottleRate == null) {
+      if (currThrottleRate == null || !Objects.equals(currThrottleRate.value(), String.valueOf(_throttleRate))) {
```

Comment:
I think Objects.equals essentially compare the references of 2 objects. I feel Objects.equals(currThrottleRate.value(), String.valueOf(_throttleRate)) will always return false since they are different objects. Please double confirm. 

For string compare, I would recommend just use currThrottleRate.value().equals(String.valueOf(_throttleRate))

Reply:
👋 The string compare is tested in example 2 in the PR description where the `leader.replication.throttled.rate` gets updated as well as the [unit test](https://github.com/linkedin/cruise-control/pull/2080/files#diff-f54134039747ea091a5603a0a8f3bd23f63efe4a8a31873ba6beb8ad0b568897R314) where the throttle rate is overwritten now. 
I also like the recommendation though so I can change that.